app/assets/javascripts/controllers/ansible_credentials/ansible_credentials_form_controller.js

```diff
@@ -40,6 +36,18 @@ ManageIQ.angular.app.controller('ansibleCredentialsFormController', ['$window',
       vm.modelCopy = angular.copy( vm.credentialModel );
       miqService.sparkleOff();
     }
+
+    var optionsPromiseResolved = optionsPromise
+      .then(getCredentialOptions)
+      .catch(miqService.handleFailure);
+
+    var dataPromiseResolved = dataPromise
```

Comment:
Sorry, `dataPromise` is undefined on http://localhost:3000/ansible_credential/new .

You have to keep the then where it was, in the if.